src/query/storage/m3/storage.go

```diff
@@ -197,6 +207,24 @@ func (s *m3storage) FetchCompressed(
 		return result, noop, err
 	}
 
+	if processor := s.adminOpts.IterationOptions().SeriesIteratorProcessor; processor != nil {
+		_, span, sampled := xcontext.StartSampledTraceSpan(ctx,
+			tracepoint.FetchCompressedInspectSeries)
+		iters := result.SeriesIterators()
+		if err := processor.InspectSeries(ctx, iters); err != nil {
+			s.logger.Error("error inspecting series", zap.Error(err))
```

Comment:
This should probably fail the query yeah? Not sure what state the iterators would be in?

Reply:
@arnikola mentioned this would likely cause an error downstream too anyhow... which is true, the original error will just be a little harder to find. 

Cleanest would be to clone entirely before handing off to processor but that's going to be very intensive and want to avoid that.